tools/make_class_power.py

```diff
@@ -0,0 +1,243 @@
+
+    #Important! Densities are in synchronous gauge!
+    pre_params['gauge'] = 'synchronous'
+
+    gparams = _build_cosmology_params(config)
+    pre_params.update(gparams)
+    redshift = config['Redshift']
+    if config['InputPowerRedshift'] >= 0:
+        redshift = config['InputPowerRedshift']
+    outputs = np.array([redshift, config['InputFutureRedshift']])
+    if extraz is not None:
+        outputs = np.concatenate([outputs, extraz])
+    #Pass options for the power spectrum
+    boxmpc = config['BoxSize'] / config['InputSpectrum_UnitLength_in_cm'] * config['UnitLength_in_cm']
+    maxk = 2*math.pi/boxmpc*config['Ngrid']*16
+    powerparams = {'output': 'dTk vTk mPk', 'P_k_max_h/Mpc' : maxk, "z_max_pk" : 1+np.max(outputs),'z_pk': outputs}
+    pre_params.update(powerparams)
+
+    #Specify an external primordial power spectrum
+    if external_pk is not None:
+        pre_params['P_k_ini'] = "external_pk"
+        pre_params["command"] = "cat ",external_pk
+
+    #Make the power spectra module
+    engine = CLASS.ClassEngine(pre_params)
+    powspec = CLASS.Spectra(engine)
+    bg = CLASS.Background(engine)
+
+    #Save directory
+    sdir = os.path.split(paramfile)[0]
+    #Get and save the transfer functions if needed
+    trans = powspec.get_transfer(z=redshift)
+    if config['DifferentTransferFunctions'] == 1.:
+        tfile = os.path.join(sdir, config['FileWithTransferFunction'])
+        save_transfer(trans, tfile, bg, redshift)
+        transfut = powspec.get_transfer(z=config['InputFutureRedshift'])
+        tfile = os.path.join(sdir, config['FileWithFutureTransferFunction'])
+        save_transfer(transfut, tfile, bg, config['InputFutureRedshift'])
+    #Get and save the matter power spectrum
+    pk_lin = powspec.get_pklin(k=trans['k'], z=redshift)
+    pkfile = os.path.join(sdir, config['FileWithInputSpectrum'])
+    if os.path.exists(pkfile):
+        raise IOError("Refusing to write to existing file: ",pkfile)
+    np.savetxt(pkfile, np.vstack([trans['k'], pk_lin]).T)
+    if extraz is not None:
+        for red in extraz:
+            trans = powspec.get_transfer(z=red)
+            tfile = os.path.join(sdir, config['FileWithTransferFunction']+"-"+str(red))
+            save_transfer(trans, tfile, bg, red)
+            #Get and save the matter power spectrum
+            pk_lin = powspec.get_pklin(k=trans['k'], z=red)
+            pkfile = os.path.join(sdir, config['FileWithInputSpectrum']+"-"+str(red))
+            if os.path.exists(pkfile):
+                raise IOError("Refusing to write to existing file: ",pkfile)
+            np.savetxt(pkfile, np.vstack([trans['k'], pk_lin]).T)
+
+def save_transfer(transfer, transferfile, bg, redshift):
```

Comment:
Better to save the velocity here as well. I find the interpolation code in C a bit hard to understand. Perhaps a next PR.

Reply:
Yes, that is next PR. This extra transfer function is not used here yet anyway.